elytron/WFCORE-5120-client-default-ssl-context.adoc

```diff
@@ -0,0 +1,86 @@
+= WFCORE-5120 Automatic registration of client side / JVM wide default SSLContext
+:author:            Diana Vilkolak Krepinska
+:email:             dvilkola@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This issue is about providing ability to register a client side JVM wide default SSLContext for the use of any libraries that support the use of the default SSL context.
+The loading of elytron client configuration should happen with minimal configuration from the user in cases where API is not used directly.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/ELY-2112[ELY-2112]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFCORE-4144[WFCORE-4144]
+* https://issues.redhat.com/browse/WFLY-13762[WFLY-13762]
+* https://issues.redhat.com/browse/WFCORE-5101[WFCORE-5101]
+
+=== Dev Contacts
+
+* mailto:dvilkola@redhat.com[Diana Vilkolak Krepinska]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+WildFly Elytron, WildFly
+
+=== Other Interested Projects
+
+NONE
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+The automatic registration will be done by implementation of new java security provider that will be added. It will provide protocol named "Default" that will load elytron client configuration and will provide initialized ssl context for clients requesting SSL context by using `SSLContext.getDefault()`.
+
+The provider will load the ssl context from either current authentication context obtained from classpath, or from authentication context obtained from file that is passed into the security provider (either programmatically or as an argument in java.security). The SSL context configured to match ALL rules is the one that will be initialized and returned by this provider.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Implementation Plan
+Implementation of new java security provider that provide protocol named "Default" which will load Elytron client configuration and will provide initialized SSL context for clients requesting SSL context by using `SSLContext.getDefault()`.
+
+== Test Plan
+
+Automatic smoke tests will be placed in wildfly-elytron repository. Each test must run in separate JVM in order to set the default SSL Context, which slows the testsuite. Because of this, only smoke tests should be put in the testsuite and other testing should be done manually.
```

Comment:
Can you separate through bullets so that is clear how many/which cases we have in here?
E.g.:

* smoke tests, will be implemented and run automatically in the WildFly Elytron repository test suite.
* some different test type (integration?) ...
* etc.

By the way, please elaborate on the slow-testsuite topic and the need for manual tests, since we know that manual tests should be just an exception. We need a very strong motivation for this.